docs/setup_installation/gcp/restrictive_permissions.md

```diff
@@ -39,16 +39,38 @@ The following permission is only needed to be able to change the head node and R
 ```
 ### Create a VPC permissions
 The following permissions are only needed if you want [managed.hopsworks.ai](https://managed.hopsworks.ai) to create VPC and subnet for you.
-If you choose an existing VPC and subnet, you can limit the permission of the Service Account by removing them. 
 
 ```yaml
 - compute.networks.create
 - compute.networks.delete
 - compute.networks.get
-...
 - compute.subnetworks.create
 - compute.subnetworks.delete
 - compute.subnetworks.get
+- compute.firewalls.create
+- compute.firewalls.delete
+```
+
+You can remove these permissions by creating your own VPC, subnet, and firewalls and selecting them during cluster creation. For the VPC to be accepted you will need to associate it with firewall rules with the following constraints:
+
+- One firewall rule associated with the VPC must allow ingress communication on all ports for communication between sources with the [service account you will attach to the cluster nodes](getting_started.md#step-3-creating-a-service-account-for-your-cluster-instances) (source and target). To create such a rule, run the following command replacing *\$NETWORK* with the name of your VPC, *\$SERVICE_ACCOUNT* with the email of your service account, and *\$PROJECT_ID* with the id of your project:
+
+```bash
+gcloud compute firewall-rules create nodetonode --network=$NETWORK --allow=all --direction=INGRESS --target-service-accounts=$SERVICE_ACCOUNT --source-service-accounts=$SERVICE_ACCOUNTT --project=$PROJECT_ID
+```
+
+- We recommend that you have a firewall rule associated with the VPC allowing TCP ingress communication to ports 443 and 80. If you don't have a rule opening port 443 the cluster will not be accessible from the internet. If you don't have a rule opening port 80 your cluster will be created with a self-signed certificate. You will have to acknowledge it by checking the *Continue with self-signed certificate* check box during [subnet selection](cluster_creation.md#step-6-vpc-and-subnet-selection). If you have no rule, at cluster creation, allowing ingress communication on specific TCP ports [managed.hopsworks.ai](https://managed.hopsworks.ai) will not be able to open or close ports on your cluster. To create this rule, run the following command replacing *\$NETWORK* with the name of your VPC, *\$SERVICE_ACCOUNT* with the email of your service account and *\$PROJECT_ID* with the id of your project:
```

Comment:
`If you have no rule, at cluster creation, allowing ingress communication on specific TCP ports [managed.hopsworks.ai](https://managed.hopsworks.ai) will not be able to open or close ports on your cluster`

I don't fully understand this sentence. How are the firewall rules associated with our permission to open/close ports?

Reply:
The thing is that a firewall rule is a protocol and a set of port that are open or closed. 
At cluster creation we check if there is at least a rule that allow communication on TCP ports (no mater the port) and if there is we get the name of this rule and add it to the cluster definition. Later on, when you want to open or close port we edit this rule. 
But if there is no such rule at cluster creation we can't take it's name and so later on we can't edit it.